lsst/ts/cbp/component.py

```diff
@@ -409,6 +410,21 @@ def get_cbp_telemetry(self):
         self.get_mask()
         self.get_mask_rotation()
 
+    def configure(self, config):
+        self.config = config
+        self._address = self.config.address
+        self._port = self.config.port
+        self.masks.mask1.name = self.config.mask1.name
+        self.masks.mask1.rotation = self.config.mask1.rotation
+        self.masks.mask2.name = self.config.mask2.name
+        self.masks.mask2.rotation = self.config.mask2.rotation
+        self.masks.mask3.name = self.config.mask3.name
+        self.masks.mask3.rotation = self.config.mask3.rotation
+        self.masks.mask4.name = self.config.mask4.name
+        self.masks.mask4.rotation = self.config.mask4.rotation
+        self.masks.mask5.name = self.config.mask5.name
+        self.masks.mask5.rotation = self.config.mask5.rotation
```

Comment:
This needlessly duplicates data, which is scary (as it can get out of sync if code alters something) and unnecessary. Please just use replace all use of `self.masks` with `self.config`.

As a very inferior alternative you could copy all the fields and not set `self.config`, but that requires many, many additional lines of code for no gain. The schema tells you exactly what will be in `config` and you can trust it.